codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/middleware/AWSSigningMiddleware.kt

```diff
@@ -22,13 +23,20 @@ import software.amazon.smithy.swift.codegen.middleware.MiddlewareRenderable
 import software.amazon.smithy.swift.codegen.middleware.MiddlewareStep
 import software.amazon.smithy.swift.codegen.model.expectTrait
 import software.amazon.smithy.swift.codegen.model.hasTrait
+import java.util.Locale
 
-typealias AWSSigningMiddlewareParamsCallback = (OperationShape) -> String
+data class AWSSigningParams(
+    val useSignatureTypeQueryString: Boolean = false,
+    val forceUnsignedBody: Boolean = false,
+    val signedBodyHeaderContentSHA256: Boolean = false,
```

Comment:
Nit: Can we name this `signBodyHeaderContentSHA256` to keep the tense consistent with the other properties? 
```suggestion
    val signBodyHeaderContentSHA256: Boolean = false,
```

Additionally, can we add comments for each of these parameters that describe what they control?

Reply:
i see now that these names mimic the values in the output string. In that case, ignore the naming change but comments would help.

Reply:
I would prefer to not have default values, they are useful but very easy to miss because not enforced and likely to result in an undesired outcome.

Reply:
Comments have been added to `AWSSigningParams` values, and default values have been removed (they were never relied on anyway.)